pkg/pipeline/encode/encode_s3.go

```diff
@@ -119,12 +118,11 @@ func (s *encodeS3) createObjectTimeoutLoop() {
 			log.Debugf("exiting createObjectTimeoutLoop because of signal")
 			return
 		case <-ticker.C:
-			log.Debugf("createObjectTimeoutLoop timer expire")
 			now := time.Now()
 			log.Debugf("time now = %v, expiryTime = %v", now, s.expiryTime)
-			if now.After(s.expiryTime) {
-				_ = s.writeObject()
-			}
```

Comment:
IIUC, this is the only place where `expiryTime` is read. If it is removed from here, maybe the variable can be removed altogether.

Reply:
This is exactly for what the parameter is needed, to define the user's desired timeout period when to create a new object. So I don't think we can remove this.